### ISSUE TITLE
Syntax highlighting on GitHub broken

### ISSUE THREAD

**Issue by zwilias** [elm-community/SublimeElmLanguageSupport#47]
Interestingly, GitHub uses this very repo as the provider for syntax highlighting on GitHub. https://github.com/github/linguist/blob/v6.0.0/.gitmodules#L679-L681

In the latest update of linguist, they updated all the submodules. The Elm module went from 581b9e6f5bfdf55e506ac5a2b18422296c375ca2 to 6f2e53603a62663463f95079b308a1c9adbabeec which is a fairly big [change](https://github.com/ThomasWeiser/Elm.tmLanguage/compare/581b9e6f5bfdf55e506ac5a2b18422296c375ca2...6f2e53603a62663463f95079b308a1c9adbabeec).

It would be pretty great if we could figure out a way forward - whether that means making this repo compatible with linguist or something else, I'll leave in the middle as I have no real idea how this bundle actually works.

**Comment by zwilias** [comment]
Yeah, Linguist essentially does both - language recognition using a built in set of rules, and based on the recognised language, highlighting using a bunch of vendored grammars.

Cool, the switch to ST3's new syntax format would indeed explain this. I'm wondering about the best next course of action. Ultimately, having a supported TextMate bundle would be best (for as long as GitHub uses that for highlighting), though I wouldn't wish that extra maintenance cost on anyone.

Perhaps it would be best to create a separate repo for a "legacy" tmLanguage definition, so at least the maintenance doesn't become your responsibility by default. 

CC/ @eeue56 do you have any thoughts on this?

**Comment by eeue56** [comment]
@zwilias I think you should open an issue like this: https://github.com/github/linguist/issues/4004 and discuss there. It might be a similar issue, in which case I would prefer us to maintain a single source of truth for syntax highlighting for Elm. If ST3 syntax highlighting isn't supported by linguist, we should indeed make a separate repo. 

I don't see anything the linguist repo explicitly stating that they require `.tmLanguage` in the docs. So it might be that they support .sublime_syntax too.

**Comment by eeue56** [comment]
I'm gonna put a temp fix in place.

**Comment by eeue56** [comment]
@zwilias Okay so I made Elm.tmLanguage again pointing to that old commit: https://github.com/elm-community/Elm.tmLanguage. Can you ask them to do a fresh pull from there?

**Comment by zwilias** [comment]
https://github.com/github/linguist/issues/4006 created. Thanks @eeue56!

**Comment by hkgumbs** [comment]
👋 I came across this issue, because I was trying to implement a small fix for GitHub syntax highlighting. I'm wondering if anyone in this conversation could clarify which repo we intend to be the source of truth for GitHub highlighting.

It seems that [GitHub will not support the new Sublime syntax](https://github.com/github/linguist/issues/3209), so keeping `Elm.tmLanguage` around specifically for GitHub seems reasonable. In that case, it might be nice to strip out the Sublime 2 stuff, to communicate to future visitors that [ st2-0.19.0](https://github.com/elm-community/SublimeElmLanguageSupport/releases/tag/st2-0.19.0) is probably what they want. GitHub _does_ support the Atom format, so telling Linguist to look there might make sense as well. Those are the only options I could think of, but I'm guessing there are more.

I'm willing to do the work, so just let me know what you all are thinking.

**Comment by sentience** [comment]
We’ll be maintaining `st2-` releases for the foreseeable future for Sublime Text 2 users, and I would say the most reasonable way to do that will be with a `st2` branch. If I set that up, GitHub Syntax Highlighting could point to that branch. Sound good, @hkgumbs?

**Comment by sentience** [comment]
Alternatively, @eeue56 if you think it makes more sense to maintain Elm.tmLanguage as a separate project, can we find it a maintainer?

**Comment by hkgumbs** [comment]
Kevin, that sounds solid to me. I didn't realize that's you were
maintaining ST2 support here. Unless we hear otherwise from others I can
look into updating linguist in the next week or two.
